ocs_ci/utility/workloadfixture.py

```diff
@@ -139,68 +141,93 @@ def prometheus_log(info, prometheus_alert_list):
+                    in constants.MANAGED_SERVICE_PLATFORMS
+                ):
+                    # During testing of ODF Managed Service are also collected alerts
+                    # in PagerDuty, Sendgrid and Dead Man's Snith systems
+                    pagerduty = PagerDutyAPI()
+                    logger.info("Logging all PagerDuty incidents")
+                    incidents_response = pagerduty.get(
+                        "incidents",
+                        payload={
+                            "service_ids[]": pagerduty_service_ids,
+                            "since": time.strftime(
+                                "%Y-%m-%d %H:%M:%S", time.gmtime(start_time)
+                            ),
+                            "time_zone": "UTC",
+                        },
+                    )
+                    incidents_response.raise_for_status()
+                    pagerduty_incidents = incidents_response.json().get("incidents")
+                    results["pagerduty_incidents"] = pagerduty_incidents
                     logger.info(
-                        f"Starting {additional_time}s sleep for the purposes of measurement."
+                        "Stopping PagerDuty periodical update of pagerduty secret"
                     )
-                    time.sleep(additional_time)
-            # Dumping measurement results into result file.
-            stop_time = time.time()
-            info["run"] = False
-            logging_thread_prometheus.join()
-            results = {
-                "start": start_time,
-                "stop": stop_time,
-                "result": result,
-                "metadata": metadata,
-                "prometheus_alerts": prometheus_alert_list,
-                "first_run": True,
-            }
-            if (
-                config.ENV_DATA["platform"].lower()
-                in constants.MANAGED_SERVICE_PLATFORMS
-            ):
-                # During testing of ODF Managed Service are also collected alerts
-                # in PagerDuty, Sendgrid and Dead Man's Snith systems
-                pagerduty = PagerDutyAPI()
-                logger.info("Logging all PagerDuty incidents")
-                incidents_response = pagerduty.get(
-                    "incidents",
-                    payload={
-                        "service_ids[]": pagerduty_service_ids,
-                        "since": time.strftime(
-                            "%Y-%m-%d %H:%M:%S", time.gmtime(start_time)
-                        ),
-                        "time_zone": "UTC",
-                    },
+                    config.RUN["thread_pagerduty_secret_update"] = "required"
+                logger.info(f"Results of measurement: {results}")
+                with open(result_file, "w") as outfile:
+                    logger.info(f"Dumping results of measurement into {result_file}")
+                    json.dump(results, outfile)
+
+        for i in range(1, 3):
```

Comment:
How do you prevent to execute the operation 3-times if oc commands work correctly?

Reply:
somehow forgot to put break after   
`try:
 do_operation()`

will it be Ok on your opinion ?

Reply:
fixed